presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1686,10 +1686,8 @@ private void executeIrreversibleMetastoreOperations()
                         synchronized (failedIrreversibleOperationDescriptions) {
                             failedIrreversibleOperationDescriptions.add(irreversibleMetastoreOperation.getDescription());
                             // A limit is needed to avoid having a huge exception object. 5 was chosen arbitrarily.
-                            synchronized (suppressedExceptions) {
```

Comment:
This is redundant not because synchronization is not needed, but because it's already surrounded by `synchronized (failedIrreversibleOperationDescriptions)`.
Maybe introduce a local `Object lock` instead of synchronizing on lists?

Reply:
> Maybe introduce a local Object lock instead of synchronizing on lists?

What's the advantage?

Reply:
Clarity of intent